### DISCUSSION TITLE
Inventree API - Option to add list of  Stock Tracking data with custom dates

### DISCUSSION THREAD

**Discussion by wieselukas** [inventree/InvenTree#7301] [Q&A]
Hi everybody,

switching from partkeepr to Inventree, I am looking to also copy over the stock history into the associating Stock Tracking entries in Inventree. Ideally this would also mean copying the date and time. Looking at the http://inventree.localhost/api-doc page, I could only find 3 get methods for the stock tracking, and no option to bulk upload Tracking info or set the dates.

I am currenly planning to use the POST /api/stock/count method and to put the partkeepr date, username and comment all into the notes field. This is a feasible option, but nevertheless: does anybody know if uploading stock tracking with dates over the api is anyhow possible?

Thanks! Lukas

**Comment by SchrodingersGat** [comment]
Hey @wieselukas welcome aboard! 

I do not think that you can do this via the API. The stock "tracking" entries are designed to be automatically generated (and thus cannot be ingested via the API with "back-dated" records).

Potentially you could do it by structuring a `.json` or `.yaml` file with the correct data and then running `invoke import-records ...` [as per the documentation](https://docs.inventree.org/en/stable/start/migrate/) - however that would require some careful consideration of how the data should be structured, and you would need to take care to back up your database before each step

**Comment by wieselukas** [comment]
Hi @SchrodingersGat ,

thanks for the info! I think I will stay with the api-based approach, as all of the import is now done through that. But interesting catch that if somebody want's to write a "full" partkeepr import script, using the api might not be a feasible option. 

Anyways I'll make sure to notify you once my import is done and share my code, so this can potentially be re-used for future switches by other people.

**Reply by SchrodingersGat** [reply]
Have you seen this project - https://github.com/rgilham/PK2InvenTree - not sure where it is at now, and I've never used partkeepr so will not be much use - but there are certainly other people out there who have looked into it.

**Reply by wieselukas** [reply]
Yes i did. I haven't found this to be enough, as e.g. companies aren't transfered. I forked another python script and am currently working on my changes on a separate branch: https://github.com/wieselukas/PartKeepr-to-InvenTree .

**Reply by wieselukas** [reply]
Heyho @SchrodingersGat,

I haven't quickly found a way to send you a direct message, so I am writing here again even though it's a bit off topic.

I just did the migration from partkeepr to Inventree for my group and thus committed all changes to the migration script's (https://github.com/wieselukas/PartKeepr-to-InvenTree) main branch. It's not exactly beautifully written, but should cover copying most Partkeepr fields and attachements now. It definetely contains much more than the pk2inventree script. 

I will add some notes to the projects Readme, but I basically think that this project could only further evolve with other people working on it while doing their own migration. I still really hope that the script can help other people migrate and thus maybe contributes a tiny bit to the whole project. 

All the best
Lukas

**Reply by SchrodingersGat** [reply]
@wieselukas glad to hear that you got the data imported :)
